src/code/dialogs/zoneDialog.js

```diff
@@ -97,18 +128,165 @@ const ZoneDialog = WDialog.extend({
           getSelectedOperation().removeZone(z.id);
         });
       }
+      if (z.points.length == 0) {
+        const addPoints = L.DomUtil.create("a", null, commandcell);
+        addPoints.textContent = " 🖍";
+        addPoints.href = "#";
+        L.DomEvent.on(addPoints, "click", (ev) => {
+          L.DomEvent.stop(ev);
+          this.ZonedrawHandler.zoneID = z.id;
+          this.ZonedrawHandler.enable();
+        });
+      } else {
+        const delPoints = L.DomUtil.create("a", null, commandcell);
+        delPoints.textContent = " 🧽";
+        delPoints.href = "#";
+        L.DomEvent.on(delPoints, "click", (ev) => {
+          L.DomEvent.stop(ev);
+          getSelectedOperation().removeZonePoints(z.id);
+        });
+      }
+      if (this.ZonedrawHandler && this.ZonedrawHandler._enabled) {
+        const stopDrawing = L.DomUtil.create("a", null, commandcell);
+        stopDrawing.href = "#";
+        stopDrawing.textcontent = " 🛑";
+        L.DomEvent.on(stopDrawing, "click", (ev) => {
+          L.DomEvent.stop(ev);
+          this.ZonedrawHandler.disable();
+        });
+      }
     }
 
-    const add = L.DomUtil.create("a", null, container);
+    /* const add = L.DomUtil.create("a", null, container);
     add.href = "#";
     add.textContent = "add zone";
     L.DomEvent.on(add, "click", (ev) => {
       L.DomEvent.stop(ev);
       getSelectedOperation().addZone();
     });
 
+    const assign = L.DomUtil.create("a", null, container);
+    assign.href = "#";
+    assign.textContent = "set markers to zones";
+    L.DomEvent.on(assign, "click", (ev) => {
+      L.DomEvent.stop(ev);
+      this.setMarkersToZones();
+    });
+    */
+
     return container;
   },
+
+  setMarkersToZones: function () {
+    const op = getSelectedOperation();
+
+    op.startBatchMode();
+    for (const m of op.markers) {
+      const ll = op.getPortal(m.portalId).latLng;
+
+      const zone = op.determineZone(ll);
+      op.setZone(m, zone);
+    }
+    op.endBatchMode();
+  },
+
+  setLinksToZones: function () {
+    const op = getSelectedOperation();
+
+    op.startBatchMode();
+    for (const l of op.links) {
+      const ll = op.getPortal(l.fromPortalId).latLng;
+      const zone = op.determineZone(ll);
+      op.setZone(l, zone);
+    }
+    op.endBatchMode();
+  },
 });
 
 export default ZoneDialog;
+
+const ZonedrawHandler = L.Handler.extend({
+  initialize: function (map = window.map, options) {
+    this.zoneID = 0;
+    this._enabled = false;
+
+    L.Handler.prototype.initialize.call(this, map, options);
+    this.options = options;
```

Comment:
```suggestion
```
 do we need `options` ?  (there is none for L.Handler)

Reply:
I copied this from quickdraw, so if it is not needed here we should check it as well